lua/telescope/builtin/files.lua

```diff
@@ -267,8 +267,105 @@ files.file_browser = function(opts)
         end
       end
 
+      local rename_file = function()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+        local old_name = Path:new(action_state.get_selected_entry()[1])
+
+        if old_name.filename == '../' then
+          print('Please select a file!')
+          return
+        end
+
+        local new_name = vim.fn.input("Insert a new name: ", old_name:make_relative())
+
+        old_name:rename({ new_name = new_name })
+        current_picker:refresh(gen_new_finder(current_picker.cwd), { reset_prompt = true })
+      end
+
+      local get_marked_files = function ()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+        local entries = current_picker:get_multi_selection()
+
+        local selected = {}
+        for _, entry in ipairs(entries) do
+          table.insert(selected, Path:new(entry[1]))
+        end
+
+        return selected
+      end
+
+      local move_file = function()
+        local current_picker = action_state.get_current_picker(prompt_bufnr)
+
+        for _, file in ipairs(get_marked_files()) do
+          local filename = file.filename:sub(#file:parents() + 2)
```

Comment:
this is what I meant by 
> this kinda works. Still need to deal with the cwd being weird like path/to/file/..

This trick works but I sometimes get `path/to/file/../../`. It only happens once and I couldn't find a way to reproduce it though.

Reply:
cwd is my bad. I stumbled across this a couple of days ago but didn't want to make a PR just for this.
You need to swap out some `vim.fn.expand` to `vim.loop.fs_realpath` because `expand` will not reduce the path.
So example `/home/conni/repos/..` will always be the same and each move will just put the new dir at the end so you might end up with `/home/conni/repos/../repos/neovim/src/..` rather than just `/home/conni/repos/neovim` let me send you a diff :)

```diff
diff --git a/lua/telescope/builtin/files.lua b/lua/telescope/builtin/files.lua
index 9749635..76d6bd8 100644
--- a/lua/telescope/builtin/files.lua
+++ b/lua/telescope/builtin/files.lua
@@ -238,7 +238,7 @@ files.file_browser = function(opts)
       action_set.select:replace_if(function()
         return action_state.get_selected_entry().path:sub(-1) == os_sep
       end, function()
-        local new_cwd = vim.fn.expand(action_state.get_selected_entry().path:sub(1, -2))
+        local new_cwd = vim.loop.fs_realpath(action_state.get_selected_entry().path:sub(1, -2))
         local current_picker = action_state.get_current_picker(prompt_bufnr)
         current_picker.cwd = new_cwd
         current_picker:refresh(gen_new_finder(new_cwd), { reset_prompt = true })
@@ -262,7 +262,7 @@ files.file_browser = function(opts)
           vim.cmd(string.format(':e %s', fpath))
         else
           Path:new(fpath:sub(1, -2)):mkdir({ parents = true })
-          local new_cwd = vim.fn.expand(fpath)
+          local new_cwd = vim.loop.fs_realpath(fpath)
           current_picker.cwd = new_cwd
           current_picker:refresh(gen_new_finder(new_cwd), { reset_prompt = true })
         end
```

Reply:
I also ran into some other issues. For example going into a symbolic link if the resolved link is a directory. Because `vim.loop.fs_scandir_next` (don't remember the function) will return either file or directory. And a link is from typ file. So we don't insert a `/` at the end so `replace_if` will return false and we run `action_set.select` :| So we might wanna change that check to, `vim.loop.fs_stat` (not `lstat` because we are not interested in the stats of the link but of the resolved one) and then check `type`

Reply:
If you want i can commit these changes to this branch, otherwise you can try to fix it yourself :)

Reply:
oh please :)
I still got some stuff to do this evening